common/src/main/java/io/netty/util/internal/DefaultPriorityQueue.java

```diff
@@ -78,6 +78,14 @@ public void clear() {
     }
 
     @Override
+    public void clearIgnoringIndexes() {
+        for (int i = 0; i < size; ++i) {
```

Comment:
If we have to iterate the array then I don't think this method is worth it. My intention for suggesting this method would be to avoid iteration of the queue. My suggestion would be we do the following:

1. don't iterate the array, and just set size to 0. (we should clarify in javadocs): it is useful in cases were GC nepotism is known not to be an issue because the queue object itself will be made eligible for GC.
2. set queue to EMPTY_ARRAY, and size to 0. the negative would be we may have to ramp up again if the queue is used after this method is called.

Option (2) is the more conservative approach. We should clarify the behavior in the javadocs, and if we need to in the future we could consider providing a "resize hint" to avoid the size ramp up overhead.

Reply:
Went with option 1 since it looks like this is only called when shutting down an event loop, so indeed the queue should always be made GC-able.